hardware/raspberrypi/spi/README.md

```diff
@@ -3,51 +3,91 @@
-- [Software](#software)
 - [Hardware](#hardware)
+- [Software](#software)
 - [Linux driver](#driver)
 - [Troubleshooting](#troubleshooting)
 
 <a name="overview"></a>
 ## Overview
 
-The Raspberry Pi is equipped with one [SPI](https://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus) bus that has 2 chip selects.
-
-The SPI master driver is disabled by default on Raspberry Pi OS. To enable it, use [raspi-config](../../../configuration/raspi-config.md), or ensure the line `dtparam=spi=on` isn't commented out in `/boot/config.txt`, and reboot. If the SPI driver was loaded, you should see the device `/dev/spidev0.0`.
-
-The SPI bus is available on the P1 Header:
-
-```
-MOSI    P1-19
-MISO    P1-21
-SCLK    P1-23   P1-24    CE0
-GND     P1-25   P1-26    CE1
-```
-
-<a name="software"></a>
-## Software
-
-### bcm2835 library
-
-This is a C library for Raspberry Pi (RPi). It provides access to GPIO and other IO functions on the Broadcom BCM 2835 chip. Accesses the hardware registers directly.
-
-http://www.airspayce.com/mikem/bcm2835/
-### Use spidev from C
-
-There's a loopback test program in the Linux documentation that can be used as a starting point. See the [Troubleshooting](#troubleshooting) section. Uses the Linux `spidev` driver to access the bus.
-
-### Shell
-
-```bash
-# Write binary 1, 2 and 3
-echo -ne "\x01\x02\x03" > /dev/spidev0.0
-```
+The Raspberry Pi family of devices is equipped with three [SPI](https://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus) buses. The main SPI interface is referred to as SPI0 in the documentation; the second is SPI1. SPI2 is only usable on a Compute Module or Compute Module 3. Devices based on BCM2711, i.e. Pi 4B and the upcoming Compute Module 4, have access to an additional 4 SPI buses (SPI3 to SPI6).
 
 <a name="hardware"></a>
 ## Hardware
 
-The BCM2835 on the Raspberry Pi has 3 SPI Controllers. Only the SPI0 controller is available on the header.
-Chapter 10 in the [BCM2835 ARM Peripherals](../bcm2835/BCM2835-ARM-Peripherals.pdf) datasheet describes this controller.
+The BCM2835 on the Raspberry Pi has 3 [http://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus SPI] Controllers. The main SPI bus (with two hardware chip selects) is available on the header of all Pis with Linux kernel support. The second SPI bus (with three hardware chip selects) is available on 40-pin versions of Pis, with kernel support from Raspbian Jessie 2016-05-10 distribution and up. The third SPI bus (also with three hardware chip selects) is only usable on a Compute Module because the pins aren't brought out onto the 40-pin header.
+
+BCM2711 adds another 4 SPI buses, each with 2 hardware chip selects. All are available on the 40-pin header (provided nothing else is trying to use the same pins).
+
+Chapter 10 in the [http://www.raspberrypi.org/wp-content/uploads/2012/02/BCM2835-ARM-Peripherals.pdf BCM2835 ARM Peripherals] datasheet describes the main controller.  Chapter 2.3 describes the auxiliary controller.
```

Comment:
Probably also worth linking to the BCM2711 datasheet? Also, IMHO it'd be better to link to the datasheet in the docs repo (i.e. `../bcm2835/BCM2835-ARM-Peripherals.pdf`) rather than linking to the one on the blog (`wp-content/uploads/...`).

"Chapter 2.3 describes the auxiliary controller." - perhaps mention which of the SPI busses apply to the auxiliary controller?

Reply:
I'll change the link.